metpy/calc/thermo.py

```diff
@@ -1444,3 +1444,61 @@ def mixed_layer(p, *args, **kwargs):
         ret.append((-1. / actual_depth.m) * np.trapz(datavar_layer, p_layer) *
                    datavar_layer.units)
     return ret
+
+
+@exporter.export
+@check_units('[pressure]', '[temperature]')
+def thickness_hydrostatic(pressure, temperature, **kwargs):
+    r"""Calculate the thickness of a layer via the hypsometric equation.
+
+    This thickness calculation uses the pressure and temperature profiles (and optionally
+    mixing ratio) via the hypsometric equation with virtual temperature adjustment
+
+    .. math:: \Delta z = -\frac{R_d}{g} \int_{p_0}^{p_1} T_v d\ln p
+
+    This assumes a hydrostatic atmosphere.
+
+    Layer bottom and depth specified in pressure.
+
+    Parameters
+    ----------
+    pressure : `pint.Quantity`
+        Atmospheric pressure profile
+    temperature : `pint.Quantity`
+        Atmospheric temperature profile
+    mixing : `pint.Quantity`, optional
+        Profile of dimensionless mass mixing ratio. Defaults to zero, so that virtual
+        temperature is simply the temperature.
+    molecular_weight_ratio : `pint.Quantity` or float, optional
+        The ratio of the molecular weight of the constituent gas to that assumed
+        for air. Defaults to the ratio for water vapor to dry air.
+        (:math:`\epsilon\approx0.622`).
+    bottom: `pint.Quantity`, optional
+        The bottom of the layer in pressure. Default is the first observation, assumed
+        to be the surface.
+    depth: `pint.Quantity`, optional
+        The depth of the layer in hPa. Defaults to full range of given pressure values.
+
+    Returns
+    -------
+    `pint.Quantity`
+        The thickness of the layer in meters.
+
+    See Also
+    --------
+    pressure_to_height_std, virtual_temperature
+
+    """
+    mixing = kwargs.pop('mixing', np.zeros_like(temperature))
+    molecular_weight_ratio = kwargs.pop('molecular_weight_ratio', epsilon)
+    bottom = kwargs.pop('bottom', None)
+    depth = kwargs.pop('depth', pressure.max() - pressure.min())
```

Comment:
I think `depth = kwargs.pop('depth', None)` should work here as well.

Reply:
Unfortunately, get_layer defaults to a depth of 100 hPa when None is specified, which didn't seem to make sense for this function (compared to the data of the whole layer). But, I can definitely still change it to that if you would like.

Reply:
Good point. Do you really need to call `get_layer` if you don’t have a `depth` argument then?

Reply:
That's a good point, I don't think I really need to in that case. So, I've made the get_layer call conditional now in my latest commit.